BSB_lan.ino

```diff
@@ -7299,8 +7299,9 @@ uint8_t pps_offset = 0;
                     case CF_MQTT:
                     case CF_MQTT_IPADDRESS:
                       if(MQTTClient){
-                        delete MQTTClient;
                         MQTTClient = NULL;
+                        delete MQTTClient;
+                        delete mqtt_client;
```

Comment:
It should be
```
                        delete MQTTClient;
                        MQTTClient = NULL;
                        delete mqtt_client;
```
because if we try "delete" with NULL address we can got something bad

Reply:
Sorry for that, in my opinion first delete and than set NULL doesn't look well.
Oh mate, there is so much to learn...

Reply:
Not sure, but doesn't have to be the `mqtt_client->stop();`come at first, before the `delete`?

Reply:
what could be a better name instead of 'mqtt_client' ?
the part 'client' should be used, but 'mqtt' also ...
What about 'client_mqtt' ?

Reply:
I just meant that the difference between MQTTClient and mqtt_client is not clear. Both say that it's a MQTT client, but apparently they have different functions, don't they?

Reply:
Ok, actually there is to use MQTTClient and client.
'mqtt_client' is a reference.
Would be 'ref_mqtt_client' better?

Reply:
Hey guys, the only open thing is renaming of the variable mqtt_client.
I don't know how I can solve it for alone.

Reply:
Sorry, now I had time to look at the code in full: I would rename MQTTClient to MQTTPubSubClient - then it's clear that one refers to the MQTT protocol and that mqtt_client refers to the network client.

Reply:
should I do this changes in my request?

Reply:
If you have the time to do so, please, otherwise I put it on my ToDo...

Reply:
Ok, I'll change it.